warehouse/integrations/azure-synapse/azure-synapse.go

```diff
@@ -953,7 +954,21 @@ func (as *AzureSynapse) LoadTable(ctx context.Context, tableName string) (*types
 func (as *AzureSynapse) Cleanup(ctx context.Context) {
 	if as.DB != nil {
 		// extra check aside dropStagingTable(table)
-		as.dropDanglingStagingTables(ctx)
+		err := as.dropDanglingStagingTables(ctx)
+		if err != nil {
+			as.logger.Warnw("Error dropping dangling staging tables",
```

Comment:
why not do this in `manager.Setup` or `New`
```
as.logger = loggerPassedDown.With(...)
```

Reply:
There's not a lot done with/by the `whManager`(synapse in this case) between `New` and `Setup`.
So setup would be an ideal candidate to assign these key value pairs to the logger.

Or like you say, some refactoring would help and this could be taken up then. It is difficult for me to follow the path and to remember the next time I visit

Reply:
makes sense, I will try to follow up with another PR here